recipes/cppcheck/all/conanfile.py

```diff
@@ -40,35 +28,40 @@ def requirements(self):
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["USE_Z3"] = self.options.with_z3
-        cmake.definitions["HAVE_RULES"] = self.options.have_rules
-        cmake.definitions["USE_MATCHCOMPILER"] = "Auto"
-        cmake.definitions["ENABLE_OSS_FUZZ"] = False
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["USE_Z3"] = self.options.with_z3
+        tc.variables["HAVE_RULES"] = self.options.have_rules
+        tc.variables["USE_MATCHCOMPILER"] = "Auto"
+        tc.variables["ENABLE_OSS_FUZZ"] = False
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def build(self):
-        self._patch_sources()
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        replace_in_file(self, os.path.join(self.source_folder, "tools", "CMakeLists.txt"), "CMAKE_SOURCE_DIR", "PROJECT_SOURCE_DIR")
+        replace_in_file(self, os.path.join(self.source_folder, "htmlreport", "cppcheck-htmlreport"), "#!/usr/bin/env python", "#!/usr/bin/env python3")
+        replace_in_file(self, os.path.join(self.source_folder, "cli", "CMakeLists.txt"),
+                              "RUNTIME DESTINATION ${CMAKE_INSTALL_FULL_BINDIR}",
+                              "DESTINATION ${CMAKE_INSTALL_FULL_BINDIR}")
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
-        self.copy("*", dst=os.path.join("bin","cfg"), src=os.path.join(self._source_subfolder,"cfg"))
-        self.copy("cppcheck-htmlreport", dst=os.path.join("bin"), src=os.path.join(self._source_subfolder,"htmlreport"))
-        cmake = self._configure_cmake()
+        copy(self, "COPYING", dst=os.path.join(self.package_folder, "licenses"), src=os.path.join(self.source_folder))
+        copy(self, "cppcheck-htmlreport", dst=os.path.join(self.package_folder, "bin"), src=os.path.join(self.source_folder, "htmlreport"))
+        cmake = CMake(self)
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "share"))
 
     def package_info(self):
+        self.cpp_info.includedirs = []
+        self.cpp_info.libdirs = []
+
         bin_folder = os.path.join(self.package_folder, "bin")
         self.output.info("Append %s to environment variable PATH" % bin_folder)
-        self.env_info.PATH.append(bin_folder)
-        # This is required to run the python script on windows, as we cannot simply add it to the PATH
-        self.env_info.CPPCHECK_HTMLREPORT = os.path.join(bin_folder, "cppcheck-htmlreport")
+        self.buildenv_info.prepend_path("PATH", bin_folder)
```

Comment:
Please, revert env_info.PATH, it's still used by Conan 1.x generators. Also you can remove this line. Buildenv_info and runenv_info automatically receive PATH.